Source/JavaScriptCore/runtime/SmallStrings.h

```diff
@@ -49,6 +50,30 @@ class StringImpl;
 
 namespace JSC {
 
+namespace SmallString {
+#define FOREACH_JS_STATIC_STRING_IMPL(macro) \
+    macro(baseConstructorString, "(function () { })") \
+    macro(derivedConstructorString, "(function (...args) { super(...args); })") \
+    macro(terminationErrorString, "JavaScript execution terminated.")
+
+#define DECLARE_STATIC_STRING_IMPL(name, characters) \
+    extern LazyNeverDestroyed<StringImpl*> s_##name; \
+    inline StringImpl* name() \
+    { \
+        return s_##name.get(); \
+    }
+FOREACH_JS_STATIC_STRING_IMPL(DECLARE_STATIC_STRING_IMPL)
+#undef DECLARE_STATIC_STRING_IMPL
+
+#define INITIALIZE_STRING(name, characters) \
+    s_##name.construct(&StringImpl::createStaticStringImplWithoutCopying(characters, sizeof(characters) / sizeof(char) - 1 /* skip null byte */).leakRef());
```

Comment:
Anywhere that calls createStaticStringImplWithoutCopying needs to either be inside a call_once, or needs a fence